### PR TITLE
DrawState test

### PR DESCRIPTION
**Pull request by davidpiuva**
Added a test that has different input and output resolution while rotating with wide angle perspective.
The old draw API would fail this test by halfway implementing local coordinates so it makes sense to assert centeredness around (0, 0) even when stretched to another aspect ratio.

### CODE REVIEW

**Comment by davidpiuva** [comment]
@emilwestergren Peer review?

File: test/draw/gpu/GpuImageScalingTest.ooc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	init: func {
+		super("GpuImageScalingTest")
+		sourceImage := RasterRgba open("test/draw/gpu/input/Flower.png")
+		this add("Scaling X rotation", func {
+			correctImage := RasterRgba open("test/draw/gpu/correct/scaling_X_rotation.png")
+			gpuImage := gpuContext createRgba(IntVector2D new(200, 150))
+			gpuImage canvas fill(ColorRgba transparent)
+			DrawState new(gpuImage) setFocalLengthNormalized(0.1f) setTransformReference(FloatTransform3D createRotationX(5.0f toRadians())) setInputImage(sourceImage) draw()
+			rasterFromGpu := gpuImage toRaster()
+			expect(rasterFromGpu distance(correctImage), is equal to(0.0f))
```
Comment (emilwestergren) [inline test/draw/gpu/GpuImageScalingTest.ooc]: `rasterFromGpu free()` ?

Comment (davidpiuva) [inline test/draw/gpu/GpuImageScalingTest.ooc]: No one does that in the tests. Should be applied to all tests then.

Comment (emilwestergren) [inline test/draw/gpu/GpuImageScalingTest.ooc]: I do

Comment (sebastianbaginski) [inline test/draw/gpu/GpuImageScalingTest.ooc]: https://github.com/magic-lang/ooc-kean/pull/1550 I'm working on it as the test code will run in android app and now it leaks much

Comment (davidpiuva) [inline test/draw/gpu/GpuImageScalingTest.ooc]: Should a leak fix to all draw tests go before or after this PR?

Comment (sebastianbaginski) [inline test/draw/gpu/GpuImageScalingTest.ooc]: This can go first, up to @marcusnaslund

Comment (marcusnaslund) [inline test/draw/gpu/GpuImageScalingTest.ooc]: We've historically been bad at memory management in tests.
The fact that other tests are bad doesn't mean this new test has to be. Add the free.

Comment (marcusnaslund) [inline test/draw/gpu/GpuImageScalingTest.ooc]: Same in the next test as well.

Comment (davidpiuva) [inline test/draw/gpu/GpuImageScalingTest.ooc]: ok

File: test/draw/gpu/GpuImageScalingTest.ooc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+use geometry
+use draw-gpu
+use draw
+use opengl
+use unit
+
+GpuImageScalingTest: class extends Fixture {
+	init: func {
+		super("GpuImageScalingTest")
+		sourceImage := RasterRgba open("test/draw/gpu/input/Flower.png")
```
Comment (marcusnaslund) [inline test/draw/gpu/GpuImageScalingTest.ooc]: Add a free to the test class that frees `sourceImage` while you're at it.

File: test/draw/gpu/GpuImageScalingTest.ooc [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		super("GpuImageScalingTest")
+		sourceImage = RasterRgba open("test/draw/gpu/input/Flower.png")
+		this add("Scaling X rotation", func {
+			correctImage := RasterRgba open("test/draw/gpu/correct/scaling_X_rotation.png")
+			gpuImage := gpuContext createRgba(IntVector2D new(200, 150))
+			gpuImage canvas fill(ColorRgba transparent)
+			DrawState new(gpuImage) setFocalLengthNormalized(0.1f) setTransformReference(FloatTransform3D createRotationX(5.0f toRadians())) setInputImage(sourceImage) draw()
+			rasterFromGpu := gpuImage toRaster()
+			expect(rasterFromGpu distance(correctImage), is equal to(0.0f))
+			(correctImage, gpuImage, rasterFromGpu) free()
```
Comment (davidpiuva) [inline test/draw/gpu/GpuImageScalingTest.ooc]: Updated

Comment (marcusnaslund) [inline test/draw/gpu/GpuImageScalingTest.ooc]: Very nice.

**Comment by marcusnaslund** [comment]
`free` shouldn't be at the top, though, but below `init` at the end of the class. Fix that and I'm happy. :)

**Comment by davidpiuva** [comment]
Updated

**Pull request merged**